src/test/java/seedu/medibook/model/UserAccountsListTest.java

```diff
@@ -0,0 +1,33 @@
+package seedu.medibook.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.medibook.model.commonfields.Name;
+import seedu.medibook.model.doctor.Doctor;
+import seedu.medibook.model.doctor.Mcr;
+
+public class UserAccountsListTest {
+
+    private UserAccountsList testUserAccountsList = new UserAccountsList();
+
+    @Test
+    public void check_invalidInput_returnEmptyOptional() {
+        Account testAccount = new Account("user", "pass",
+                new Doctor(new Name("Brandon"), new Mcr("MP2037X")));
+        testUserAccountsList.addAccount(testAccount);
+        assertEquals(Optional.empty(), testUserAccountsList.check("user2", "pass"));
+        assertEquals(Optional.empty(), testUserAccountsList.check("user", "pass2"));
+    }
+
+    @Test
+    public void check_validInput_returnOptionalOf() {
```

Comment:
Would you consider renaming this to check_validInput_successWithCorrectAccount()?